### PR TITLE
Fmt control

### PR DESCRIPTION
**Pull request by Akatsuki57**
More fmt::format instead of sprintf
cLog use now fsteam

### CODE REVIEW

File: drawers/cMessageDrawer.h [hunk 1]
```diff
@@ -32,15 +34,15 @@ class cMessageDrawer {
...
+		void setMessage(const std::string &msg);  // set a new message
+		const std::string& getMessage() { return cMessage; }
+		bool hasMessage() { return (!cMessage.empty()); }
 
 	protected:
 
 	private:
 		// MESSAGING / HELPER
-		char cMessage[255];
+		std::string cMessage;
```
Comment (pbomta) [inline drawers/cMessageDrawer.h:45]: 👍

File: cGame_logic.cpp [hunk 2]
```diff
@@ -1623,19 +1623,10 @@ void cGame::drawCombatMouse() {
...
-    } else if (m_screenshot < 100) {
-        sprintf(filename, "%dx%d_00%d.bmp", m_screenX, m_screenY, m_screenshot);
-    } else if (m_screenshot < 1000) {
-        sprintf(filename, "%dx%d_0%d.bmp", m_screenX, m_screenY, m_screenshot);
-    } else {
-        sprintf(filename, "%dx%d_%d.bmp", m_screenX, m_screenY, m_screenshot);
-    }
 
-    save_bmp(filename, bmp_screen, general_palette);
+    std::string filename = fmt::format("{}x{}_{:0>4}.bmp", m_screenX, m_screenY, m_screenshot);
```
Comment (pbomta) [inline cGame_logic.cpp:1627]: 👍

File: gamestates/cSelectYourNextConquestState.cpp [hunk 3]
```diff
@@ -111,10 +111,8 @@ void cSelectYourNextConquestState::thinkFast() {
         }
 
         cMessageDrawer *pDrawer = drawManager->getMessageDrawer();
-        char *cMessage = pDrawer->getMessage();
-
         isFinishedConqueringRegions = true;
-        bool isNotDisplayingMessage = cMessage[0] == '\0';
+        bool isNotDisplayingMessage = pDrawer->hasMessage();
```
Comment (pbomta) [inline gamestates/cSelectYourNextConquestState.cpp:115]: Logic inverterted here accidentally.
```suggestion
        bool isNotDisplayingMessage = !pDrawer->hasMessage();
```
or perhaps
```suggestion
        bool isDisplayingMessage = pDrawer->hasMessage();
```
And then invert the logic that uses the boolean.

File: utils/cLog.cpp [hunk 4]
```diff
@@ -165,12 +165,12 @@ void cLogger::log(eLogLevel level, eLogComponent component, const std::string& e
...
     logline += event;
 
-    fprintf(m_file, "%s\n", logline.c_str()); // print the text into the file
-    fflush(m_file);
+    logCommentLine(logline);
 }
 
 void cLogger::logCommentLine(const std::string& txt) {
-    fprintf(m_file, "\\\\%s\n", txt.c_str()); // print the text into the file
+    m_file << txt << std::endl;
```
Comment (pbomta) [inline utils/cLog.cpp:172]:
```suggestion
    m_file << "\\" << txt << std::endl;
```
This is some feature that Stefan likes. There are lines like this in the logfile:
```
\\-------
\\Allegro
\\-------
```

File: utils/cLog.cpp [hunk 5]
```diff
@@ -165,12 +165,12 @@ void cLogger::log(eLogLevel level, eLogComponent component, const std::string& e
 
     logline += event;
 
-    fprintf(m_file, "%s\n", logline.c_str()); // print the text into the file
-    fflush(m_file);
+    logCommentLine(logline);
```
Comment (pbomta) [inline utils/cLog.cpp:168]: The comment line function is to log text with two backslaches in front of it.
TBH, I'm not a big fan of it, but Stefan likes it to have "headers" before certain blocks of log lines.
What do you say, @stefanhendriks, remove this feature? We log the component with each log line, that may be enough.
I'm OK either way.

File: utils/cLog.h [hunk 6]
```diff
@@ -65,7 +68,8 @@ class cLogger {
       cLogger(const cLogger&) = delete;
       cLogger(cLogger&&) = delete;
 
-      std::FILE* m_file;
+      //std::FILE* m_file;
```
Comment (pbomta) [inline utils/cLog.h:71]: Let's just remove the commented out lines.

**Review by pbomta — CHANGES_REQUESTED** [review]
Nice improvement!
If the logic is fixed and the comment line reinstated (or removed, if Stefan is OK with that), then it can go in.

File: utils/cLog.h (hunk 6 above)
Comment (Akatsuki57) [reply utils/cLog.h]: Done

File: utils/cLog.cpp (hunk 5 above)
Comment (Akatsuki57) [reply utils/cLog.cpp]: I fix as previously

File: utils/cLog.cpp (hunk 4 above)
Comment (Akatsuki57) [reply utils/cLog.cpp]: Oups. i fix as previously

**Review by pbomta — APPROVED** [review]

**Comment by Akatsuki57** [comment]
This branch compile but she is unstable. I will check what's wrong !

File: utils/cLog.cpp (hunk 4 above)
Comment (stefanhendriks) [reply utils/cLog.cpp]: haha true. I like to have some 'headers' to tell when certain parts of logging start... so yeah please keep that ;-)

**Comment by stefanhendriks** [comment]
nice work @Akatsuki57 ! Let us know when it is stable. I'll test it then on my machine and when it works I'll approve / merge.

**Comment by stefanhendriks** [comment]
I made a draft of this PR so we know the status. If it is ready to go we can put it back with "Ready for review"

**Pull request closed without merging**